test/zdtm_ct.c

```diff
@@ -83,7 +83,15 @@ int main(int argc, char **argv)
 		return 1;
 	pid = fork();
 	if (pid == 0) {
-		if (create_timens())
+		/*
+		 * FIXME(issues/1266): Run tests in the host time namespace.
+		 * start_time in /proc/pid/stat is printed in the host time
+		 * namespace, but  /proc/uptime is shown in the current time
+		 * namespace, so we can't compare them to detect tasks that
+		 * reuse old pids. This is a kernel issue which has to be fixed
+		 * in the 5.10 kernel.
+		 */
+		if (0 && create_timens())
```

Comment:
@avagin  what do you think about something like the following?
```diff
diff --git a/test/zdtm_ct.c b/test/zdtm_ct.c
index 5495d61eb..b050eb7a7 100644
--- a/test/zdtm_ct.c
+++ b/test/zdtm_ct.c
@@ -9,6 +9,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <linux/version.h>
 
 #ifndef CLONE_NEWTIME
 #define CLONE_NEWTIME   0x00000080      /* New time namespace */
@@ -42,6 +43,8 @@ static inline int _settime(clockid_t clk_id, time_t offset)
 
 static int create_timens()
 {
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,10,0)
        int fd;
 
        if (unshare(CLONE_NEWTIME)) {
@@ -65,6 +68,7 @@ static int create_timens()
        if (setns(fd, 0))
                exit(1);
        close(fd);
+#endif
 
        return 0;
 }
```

Reply:
Hmm, maybe something like this?
```
diff --git a/test/zdtm_ct.c b/test/zdtm_ct.c
index 5495d61eb7c6..54d37b06c4d0 100644
--- a/test/zdtm_ct.c
+++ b/test/zdtm_ct.c
@@ -9,6 +9,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <sys/utsname.h>
 
 #ifndef CLONE_NEWTIME
 #define CLONE_NEWTIME   0x00000080      /* New time namespace */
@@ -42,7 +43,22 @@ static inline int _settime(clockid_t clk_id, time_t offset)
 
 static int create_timens()
 {
+       struct utsname buf;
+       unsigned major, minor;
        int fd;
+       int ret;
+
+       ret = uname(&buf);
+       if (ret)
+               return -1;
+
+       if (sscanf(buf.release, "%u.%u", &major, &minor) != 2)
+               return -1;
+
+       if (major == 5 && minor < 10) {
+               fprintf(stderr, "timens isn't supported on %s\n", buf.release);
+               return 0;
+       }
 
        if (unshare(CLONE_NEWTIME)) {
                if (errno == EINVAL) {
```

On my not-yet updated kernel:
```
[dima@Mindolluin test]$ sudo ./zdtm.py run -t zdtm/static/busyloop00 -f h -k always; sudo chown -R dima dump/
userns is supported
=== Run 1/1 ================ zdtm/static/busyloop00
timens isn't supported on 5.9.6-arch1-1
======================= Run zdtm/static/busyloop00 in h ========================
Start test
./busyloop00 --pidfile=busyloop00.pid --outfile=busyloop00.out
Run criu dump
Run criu restore
Send the 15 signal to  52
Wait for zdtm/static/busyloop00(52) to die for 0.100000
======================= Test zdtm/static/busyloop00 PASS =======================
```

Reply:
I have merged this into my patch. Thanks.